pkg/aws/aws.go

```diff
@@ -681,19 +899,24 @@ func Create(
 	ctx context.Context,
 	cfg aws.Config,
 	providerAws *AwsProvider,
-) (*ec2.RunInstancesOutput, error) {
+) (Machine, error) {
 	svc := ec2.NewFromConfig(cfg)
 
 	devpodSG, err := GetDevpodSecurityGroups(ctx, providerAws)
 	if err != nil {
-		return nil, err
+		return Machine{}, err
 	}
 
 	volSizeI32 := int32(providerAws.Config.DiskSizeGB)
 
 	userData, err := GetInjectKeypairScript(providerAws.Config.MachineFolder)
 	if err != nil {
-		return nil, err
+		return Machine{}, err
+	}
+
+	r53Zone, err := GetDevpodRoute53Zone(ctx, providerAws)
```

Comment:
What do you think of adding an extra option for the Route53 label:value lookup?
Right now all users pay the "look through route53"-tax regardless whether they want/need to use it or not.
Something simple like `AWS_USE_ROUTE53` would be enough

Reply:
it is definitely reasonable to add an extra option. I see two possible approaches:

1. Having an explicit `AWS_USE_ROUTE53` option like you suggested. This would mean, the `AWS_ROUTE53_ZONE_NAME` is then also only evaluated when `AWS_USE_ROUTE53=true`. This makes it very explicit but needs more config. 
2. We could also use `AWS_ROUTE53_ZONE_NAME` as is and add a flag called `AWS_USE_ROUTE53_ZONE_LOOKUP`, which would enable the query for the zone with the DevPod tag. In this case Route53 hostname feature would be enabled either by one or the other option. 

I personally like option 2) a bit more because it is obvious what is happening. WDYT?

Reply:
However, I see there is `AWS_USE_INSTANCE_CONNECT_ENDPOINT` and `AWS_USE_SESSION_MANAGER` with boolean values. I think I'll implement 1) to be consistent.

Reply:
Implemented option 2), see https://github.com/loft-sh/devpod-provider-aws/pull/36/commits/b89ce751a2a70865ebb93e9ba50367e2777fa943